app/src/main/java/com/travelguide/activities/TravelGuideActivity.java

```diff
@@ -498,4 +579,51 @@ public void onLoginOrLogout(boolean status) {
         setHeaderProfileInfo(false);
     }
 
+    private void getSharedPreferences() {
+//        SharedPreferences userInfo = getSharedPreferences("userInfo", 0);
+//        name = userInfo.getString("name", "missing");
+//        email = userInfo.getString("email", "missing");
+//        profilePicUrl = userInfo.getString("profilePicUrl", "missing");
+//        coverPicUrl = userInfo.getString("coverPicUrl", "missing");
+        name = Preferences.readString(this, Preferences.User.NAME);
+        email = Preferences.readString(this, Preferences.User.EMAIL);
+        profilePicUrl = Preferences.readString(this, Preferences.User.PROFILE_PIC_URL);
+        coverPicUrl = Preferences.readString(this, Preferences.User.COVER_PIC_URL);
+    }
+
+    private void loadBackdrop() {
+        getSharedPreferences();
+        if (!name.equals(Preferences.DEF_VALUE))
```

Comment:
Doing it in the opposite could avoid NullPointerException because Preferences.DEF_VALUE will never be null.

!Preferences.DEF_VALUE.equals(name)

If you are trying to check if a String reference is equal to another string that you know is not a null reference, then do something like this:

String x = "this is not a null reference"
if (x.equalsIgnoreCase(yourStringReferenceThatMightBeNull) ) {
  //do fun stuff
}
If there is any doubt as to whether or not you have null references for both Strings you are comparing, you'll need to check for a null reference on at least one of them to avoid the possibility of a NullPointerException.